packages/cli/api-importers/openapi/openapi-ir-to-fern/src/buildEndpointExample.ts

```diff
@@ -35,24 +41,49 @@ export function buildEndpointExample({
     }
 
     const hasEndpointHeaders = endpointExample.headers != null && endpointExample.headers.length > 0;
-    const hasGlobalHeaders = Object.keys(context.builder.getGlobalHeaders()).length > 0;
+    const globalHeaders = context.builder.getGlobalHeaders();
+    const hasGlobalHeaders = Object.keys(globalHeaders).length > 0;
 
     const endpointHeaderNames = new Set(endpointExample.headers?.map((header) => header.name) ?? []);
 
+    // generate examples for headers, but ensure there are no duplicates from global and auth headers
     if (hasEndpointHeaders || hasGlobalHeaders) {
-        const namedFullExamples: NamedFullExample[] = [
-            // ignore auth headers
-            ...(endpointExample.headers ?? []).filter((header) => header.name !== context.builder.getAuthHeaderName()),
-
-            // include global headers that are not already in the endpoint headers
-            ...(Object.entries(context.builder.getGlobalHeaders())
-                .filter(([header]) => !endpointHeaderNames.has(header))
-                .map(([header, _]) => ({
-                    name: header,
-                    value: FullExample.primitive(PrimitiveExample.string(header))
-                })) ?? [])
-        ];
-        // create examples for all headers
+        const namedFullExamples: NamedFullExample[] = [];
+
+        // ignore auth headers
+        if (endpointExample.headers != null) {
+            for (const header of endpointExample.headers) {
+                if (header.name !== context.builder.getAuthHeaderName()) {
+                    namedFullExamples.push(header);
+                }
+            }
+        }
+
+        // include global headers that are not already in the endpoint headers
+        for (const [header, info] of Object.entries(globalHeaders)) {
+            if (!endpointHeaderNames.has(header)) {
+                // get 'literal' headers and use the value as the example
+                if (typeof info === "object" && "type" in info && info.type.startsWith("literal")) {
```

Comment:
recursivelyVisitRawTypeReference and handle the literal case, otherwise just use `noop` from core utitlies on the other cases